scripts/jenkins/ardana/gerrit/gerrit.py

```diff
@@ -0,0 +1,235 @@
+    r"^\W*Depends-On: ("
+    r"((http(s)?:\/\/)?gerrit\.suse\.provo\.cloud\/(\/)?(\#\/c\/)?(\d+).*?)"
+    r"|(I[0-9a-f]{40})"
+    r")\s*$",
+    re.MULTILINE | re.IGNORECASE)
+
+
+def find_dependency_headers(message):
+    # Search for Depends-On headers
+    dependencies = []
+
+    for match in DEPENDS_ON_RE.findall(message):
+        # Grab out the change-id
+        if match[6]:
+            # Group 6 is the change-id matcher from the URL
+            # NOTE: Because we only pull out the change-id here, if a patchset
+            #       was included in the URL it will be ignored at this point.
+            #       The check later on will be missed and thus no warning is
+            #       raised.
+            dependencies.append(match[6])
+        elif match[7]:
+            # Group 7 is the gerrit ID in format Ia32...234a (41 chars)
+            dependencies.append(match[7])
+    # Reverse the order so that we process the oldest Depends-On first
+    dependencies.reverse()
+    return dependencies
+
+
+class GerritChange:
+    """
+    Holds the state of a gerrit change
+    """
+
+    @staticmethod
+    def _query_gerrit(query):
+        query_url = GERRIT_URL + query
+        print("Running query %s" % query_url)
+        response = requests.get(query_url, verify=GERRIT_VERIFY)
+        print("Got response: %s" % response)
+        return json.loads(response.text.replace(")]}'", ''))
+
+    def __eq__(self, other):
+        return self.id == other.id
+
+    def __hash__(self):
+        return hash(self.id)
+
+    def __init__(self, change_id=None, branch=None,
+                 patchset=None, change_object=None):
+        if change_id:
+            print("Processing given change id: %s" % change_id)
+            if change_id.isdigit():
+                self._get_numeric_change(change_id, patchset)
+            elif change_id.split('/')[0].isdigit():
+                change_id, patchset = change_id.split('/')
+                self._get_numeric_change(change_id, patchset)
+            elif change_id.startswith('I') and len(change_id) == 41:
+                self._get_change_id(change_id, branch, patchset)
+            else:
+                raise Exception("Unknown change id format (%s)" % change_id)
+        else:
+            self._set_change_object(change_object)
+
+        # Take the known self._change_object and load the attributes we want
+        self._load_change_object(patchset)
+        self._dependencies = []
+
+    def _get_numeric_change(self, change_id, patchset=None):
+        """
+        Get a change object from a deterministic numeric change number
+        """
+        query = '/changes/{}/'.format(change_id)
+        if patchset:
+            query += '?o=ALL_REVISIONS&o=ALL_COMMITS&o=DETAILED_LABELS'
+        else:
+            query += '?o=CURRENT_REVISION&o=CURRENT_COMMIT&o=DETAILED_LABELS'
+        response_json = self._query_gerrit(query)
+
+        self._change_object = response_json
```

Comment:
since the helper exists, we should use _set_change_object() here in case we do further processing the in the future